runtime/vm/jvminit.c

```diff
@@ -7283,7 +7283,7 @@ protectedInitializeJavaVM(J9PortLibrary* portLibrary, void * userData)
 		IDATA disabled = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXPRINTFLAGSFINALDISABLE, NULL);
 		if (enabled > disabled) {
 			size_t maxHeapSize = (size_t) (vm->memoryManagerFunctions->j9gc_get_maximum_heap_size(vm));
-			uint64_t maxDirectMemorySize = (uint64_t) (vm->directByteBufferMemoryMax);
+			uint64_t maxDirectMemorySize = (uint64_t) (~(UDATA)0 == (vm->directByteBufferMemoryMax) ? 0 : vm->directByteBufferMemoryMax);
```

Comment:
I don't see the parentheses I requested for `==`, nor the ones we should have for the `~` expression:
```c
			uint64_t maxDirectMemorySize = (uint64_t) (((~(UDATA)0) == vm->directByteBufferMemoryMax) ? 0 : vm->directByteBufferMemoryMax);
```